Hotkeys/Fish Water Elemental.lua

```diff
@@ -1,61 +1,45 @@
 init start
-	-- local SCRIPT_VERSION = '1.1.1'
-
-	-- If set to true, will pause walking and wait for fresh bodies to become
-	-- 'fisable'.
-	local waitFresh = false
-
+	-- local SCRIPT_VERSION = '1.1.2'
 
 	-- DO NOT EDIT BELOW THIS LINE --
-
 	local fishLoot = {281, 282, 3026, 3029, 3032, 9303}
 
-	-- I wish I could have made this a more generic function, but it would most
-	-- likely make it less efficient for the current use case; I know... I will
-	-- pay for that in the future. Meh... whatever, bring it on!
 	local function findwaterfishspots(exceptClose)
-		local hasFresh = false
-
 		for x = -7, 7 do
 			for y = -5, 5 do
-				-- This lets us ignore bodies that are too close and might fuck
-				-- up because of character's first movements, but still account
-				-- for them after it's already standing.
 				if not exceptClose or (math.abs(x) > 1 or math.abs(y) > 1) then
 					local curX, curY = $posx + x, $posy + y
 					local item = topuseonitem(curX, curY, $posz).id
 
-					if item == 9582 then
+					if table.find({9582, 4037}, item) then
 						return {x = curX, y = curY}
-					elseif item == 4037 then
-						hasFresh = true
 					end
 				end
 			end
 		end
 
-		return nil, hasFresh
+		return nil
 	end
 init end
 
 auto(100)
 
-if $lootsaround == 0 and $targetingtarget.hppc == 0 then
-	local old, fresh = findwaterfishspots(true)
+if $lootsaround == 0 and $targetingtarget.hppc == 0 and (clientitemhotkey(3483, 'crosshairs') ~= 'not found' or $fasthotkeys) then
```

Comment:
Most people put the rod into loot backpack (for revealing items inside this bp). So after opening next backpack inside, it stays invisible for bot.